src/corecel/cont/Bitset.hh

```diff
@@ -282,12 +286,16 @@ Bitset<N>::operator[](size_type pos) noexcept -> reference
 
 //---------------------------------------------------------------------------//
 //! Get the value of the bit at position pos. Equivalent to operator[] with
-//! a precondition check in debug mode.
+//! a precondition check.
 template<size_type N>
 CELER_CONSTEXPR_FUNCTION bool Bitset<N>::test(size_type pos) const
-    noexcept(!CELERITAS_DEBUG)
 {
-    CELER_EXPECT(pos < N);
+    if (CELER_UNLIKELY(pos >= N))
+    {
+        throw std::out_of_range("Bitset out of bounds: pos="
```

Comment:
Is this device compatible? I don't need to strictly hold to the standard behavior wrt exceptions... (And I guess it wouldn't be constexpr on device because that could use non-constexpr functions `assert` and `__trap`)?

Reply:
It probably doesn't make sense to call this on device in the first place; maybe we can remove it or make it host-only?

Reply:
Maybe we should just remove it then? I still don't get why we'd treat the bounds checking differently in this case than for our other quasi-standard-library classes Span and Array.

Reply:
Because the only purpose of that `test` function is to have always-on bound checks, otherwise, there is an equivalent `bool operator[]` where we can use the usual `CELER_EXPECT` bound check. I have updated it to make it device-compatible by using `assert` but am fine removing it if you think it's better.

Reply:
I gotcha. Let's try first to be consistent with our other utilities: our `array` doesn't have an `at` function, and generally we always bound check if debug, never if not. So perhaps just delete this function?

Reply:
Okay, I'll also add a `CELER_EXPECT` where needed